portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +26,31 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    public List<String> messages;
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Ciara!");
-  }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String json = convertToJsonUsingGson(messages);
+
+        response.setContentType("application/json;");
+        response.getWriter().println(messages);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String comment = request.getParameter("box");
+        messages.add(comment);
+
+        response.sendRedirect("/index.html");
+    }
+
+    public void init(){
+        messages = new ArrayList<>();
+    }
+
+    private String convertToJsonUsingGson(List<String> messages) {
+        Gson gson = new Gson();
+        String json = gson.toJson(messages);
+        return json;
+    }
```

Comment:
This logic here is too simple to worth being a method.  Perhaps make it " fetch data from datastore, sort and return comments".